demo

```diff
@@ -133,28 +137,59 @@ docker-compose \
     -f docker-compose.yml \
     -f docker-compose.storage.minio.yml \
     -f docker-compose.demo.yml \
+    -p ${DOCKER_COMPOSE_PROJECT_NAME} \
     $CLIENT \
     "$@" -d
 
+retval=$?
+if [[ $retval -ne 0 ]]; then
+    echo "Failed to start docker compose"
+    exit $retval
+fi
+
 USER='mender-demo@example.com'
-PASSWORD=$(< /dev/urandom 2>/dev/null tr -dc A-Za-z0-9 | head -c12)
+PASSWORD=$(base64 /dev/urandom | head -c 12)
 
-MENDER_SERVER_USER="${USER}:${PASSWORD}"
-MENDER_SERVER_URI="https://localhost"
 
 RETRY_LIMIT=5
 
+# Get the IP of the mender-api-gateway
+RETRIES=0
+while :
+do
+    MENDER_TEST_GATEWAY=$(docker ps -q --filter name=${DOCKER_COMPOSE_PROJECT_NAME}_mender-api-gateway | xargs docker inspect --format={{.NetworkSettings.Networks.${DOCKER_COMPOSE_PROJECT_NAME}_mender.IPAddress}})
+    if [[ $MENDER_TEST_GATEWAY == "" ]]; then
+        echo "The mender-gateway-api service does not seem to be up yet. Retrying..."
+    else
+        echo "Mender gateway IP address: " $MENDER_TEST_GATEWAY
+        break
+    fi
+    if [[ RETRIES -eq 5 ]]; then
+        echo "Failed to get the IP of the 'mender-api-gateway' service"
+        exit 1
+    fi
+    RETRIES=$((RETRIES+1))
+    sleep 5
+done
+
+MENDER_SERVER_URI="https://${MENDER_TEST_GATEWAY:-localhost}"
+
+# On MacOS docker is run in a VM,
+# therefore curl calls to the container
+# IP address does not reach the containers
+# like they should. Therefore, default to localhost.
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    MENDER_SERVER_URI="https://localhost"
+fi
+
 # Block until the useradm service returns an HTTP 4xx response
 echo "Waiting for the useradm service to start up..."
 RETRIES=0
 while :
 do
-    curl --silent -k -X POST -u $MENDER_SERVER_USER\
+    curl --silent -k -X POST -u ${USER}:${PASSWORD} \
          --fail\
-         --connect-timeout 120\
-         --retry 60\
-         --retry-delay 2\
-         --retry-max-time 120\
+         --connect-timeout 5\
```

Comment:
Did you mean to reduce it to 5? It's quite short.

There are a couple of these.

Reply:
Yes, I did. Since it is already retried 5 times, that means 5x5 seconds is spent on each phase. But you are a much better judge of the default values than me. What would you like to see?

Reply:
Ah, there is also the retries. I forgot about that. Then it's fine, I think! :-)